### PR TITLE
[cdc] Added database action factory base class.

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Added database action factory base class.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SyncDatabaseActionFactoryBase.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import static org.apache.paimon.flink.action.cdc.CdcActionCommonUtils.TABLE_PREFIX;
+import static org.apache.paimon.flink.action.cdc.CdcActionCommonUtils.TABLE_SUFFIX;
+import static org.apache.paimon.flink.action.cdc.CdcActionCommonUtils.TYPE_MAPPING;
+
+/** Base {@link ActionFactory} for synchronizing into database. */
+public abstract class SyncDatabaseActionFactoryBase<T extends SyncDatabaseActionBase>
+        extends SynchronizationActionFactoryBase<T> {
+
+    protected String warehouse;
+    protected String database;
```
Comment (TaoZex) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SyncDatabaseActionFactoryBase.java:38]: How about use `final`?

Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SyncDatabaseActionFactoryBase.java:38]: The subclass of ActionFactory is an instance generated through dynamic discovery through spi. This ActionFactory can only have a parameterless constructor, so the variable here does not use `final`.

Comment (TaoZex) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SyncDatabaseActionFactoryBase.java:38]: Thanks for your responce. Overall look good to me.

**Review by yuzelin — APPROVED** [review]
+1

**Comment by zhuangchong** [comment]
Thanks. @yuzelin

**Pull request merged**